Sources/SwiftSyntaxMacroExpansion/MacroExpansion.swift

```diff
@@ -335,6 +343,38 @@ public func expandAttachedMacroWithoutCollapsing<Context: MacroExpansionContext>
         $0.formattedExpansion(definition.formatMode)
       }
 
+    case (let attachedMacro as PreambleMacro.Type, .preamble):
+      guard let declToPass = Syntax(declarationNode).asProtocol(SyntaxProtocol.self) as? (DeclSyntaxProtocol & HasTrailingOptionalCodeBlock)
+      else {
+        // Compiler error: declaration must have a body.
+        throw MacroExpansionError.declarationHasNoBody
+      }
+
+      let preamble = try attachedMacro.expansion(
+        of: attributeNode,
+        providingPreambleFor: declToPass,
+        in: context
+      )
+      return preamble.map {
+        $0.formattedExpansion(definition.formatMode)
+      }
+
+    case (let attachedMacro as BodyMacro.Type, .body):
```

Comment:
Hmmm, this doesn't trigger if the function we're attaching the macro to has a body, but that's the case we'd want -- am I confused about something here or are we talking about different kinds of macros perhaps?